classes/sushi/CounterR5Report.inc.php

```diff
@@ -0,0 +1,537 @@
+                    $this->granularity = $attribute['Value'];
+                    break;
+            }
+        }
+    }
+
+    /**
+     * Get report items
+     */
+    abstract public function getReportItems(): array;
+
+    protected function addWarning(array $exception): void
+    {
+        $this->warnings[] = $exception;
+    }
+
+    /**
+     * Process report parameters
+     *
+     * @throws SushiException
+     */
+    public function processReportParams($request, $params): void
+    {
+        $this->context = $request->getContext();
+
+        $this->checkRequiredParams($params);
+
+        $this->checkCustomerId($params);
+
+        $this->checkDate($params);
+
+        $this->setPlatform($request->getSite());
+
+        $this->checkSupportedParams($params);
+
+        $this->checkFilters($params);
+
+        $this->checkAttributes($params);
+    }
+
+    /**
+     * Set the platform name and ID
+     */
+    protected function setPlatform($site): void
+    {
+        $platformName = $this->context->getName($this->context->getPrimaryLocale());
+        $platformId = $this->context->getPath();
+        if ($site->getData('siteSushiPlatform')) {
+            if ($site->getData('title')) {
+                $platformName = $site->getTitle($site->getPrimaryLocale());
+            }
+            $platformId = $site->getData('siteSushiPlatformID');
+        }
+        $this->platformName = $platformName;
+        $this->platformId = $platformId;
+    }
+
+    /**
+     * Check if the required parameter are provided
+     *
+     * @throws SushiException
+     */
+    protected function checkRequiredParams($params): void
+    {
+        $missingRequiredParams = [];
+        if (!isset($params['customer_id'])) {
+            $missingRequiredParams[] = 'customer_id';
+        }
+        if (!isset($params['begin_date'])) {
+            $missingRequiredParams[] = 'begin_date';
+        }
+        if (!isset($params['end_date'])) {
+            $missingRequiredParams[] = 'end_date';
+        }
+        if (!empty($missingRequiredParams)) {
+            throw new SuhiException(
+                'Insufficient Information to Process Request',
+                1030,
+                'Fatal',
+                __('sushi.exception.1030.missing', ['params' => implode(__('common.commaListSeparator'), $missingRequiredParams)]),
+                400
+            );
+        }
+    }
+
+    /**
+     * Check if the customer ID is valid
+     *
+     * @throws SushiException
+     */
+    protected function checkCustomerId($params): void
+    {
+        $institutionName = $institutionId = null;
+        $customerId = $params['customer_id'];
+        if (is_numeric($customerId)) {
+            if ($customerId == 0) {
+                $institutionName = 'The World';
+            } else {
+                $institution = Repo::institution()->get($customerId);
+                if (isset($institution) && $institution->getContextId() == $this->context) {
```

Comment:
I think `$this->context` will be an object, not the ID.

Reply:
(Adding typehints throughout will help catch these at the source!)

Reply:
> (Adding typehints throughout will help catch these at the source!)

Here we can not add extra/additional typehints, right?
institution->getContextId() does return int (typehinted)... :thinking: